### PR TITLE
added FE of Profile page

### PR DESCRIPTION
**Pull request by kqpham**
Added functionality to prepare for contest/ submission integration

### What this PR does (required):
- Completed FE profile page to match up to mockup
- Added in handlers for contest integration

### Screenshots / Videos (required):
- 

https://user-images.githubusercontent.com/17209997/120907829-cd4d6d00-c632-11eb-9945-5272ead14c90.mov

### Any information needed to test this feature (required):
- Post what steps to take to be able to test this feature (eg. "sign up as a new user and upload a profile picture from the navbar")
- Logged into an existing user, went into profile page to display contests that related to the user.
- went between the tabs to see the filtered data.
- tested the links to ensure proper routes.
- Logged into a user that has no contest connected to view alternate message in place of contest cards

### Any issues with the current functionality (optional):
- List any problems where you were not able to complete all tasks on ticket
- List any problems this PR may cause for the project or other active PRs
- Post a screenshot/video of the problem, along with detailed console outputs where applicable

### CODE REVIEW

File: client/src/components/ListView/ListView.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+                  <Grid
+                    className={classes.ListItems}
+                    container
+                    direction="column"
+                    justify="flex-start"
+                    alignItems="flex-start"
+                  >
+                    <ListItemText
+                      primary={
+                        <React.Fragment>
```
Comment (bonnieli) [inline client/src/components/ListView/ListView.tsx:37]: do we need this extra Fragment?

File: client/src/pages/Profile/Profile.tsx [hunk 2]
```diff
@@ -1,7 +1,132 @@
...
+
+interface Props {
+  loggedIn: boolean;
+  user: User;
+}
+
+export default function Profile({ user }: Props): JSX.Element {
+  const classes = useStyles();
+  const [value, setValue] = useState(0);
+  const conts = contests;
```
Comment (bonnieli) [inline client/src/pages/Profile/Profile.tsx:30]: I would name this something different, conts seems not very meaningful

File: client/src/pages/Profile/Profile.tsx [hunk 3]
```diff
@@ -1,7 +1,132 @@
...
+            <Typography>{children}</Typography>
+          </Box>
+        )}
+      </div>
+    );
+  }
+
+  //Reduces the collection of contest to contests that involves the user
+  const handleContest = () => {
+    setNewContest(conts.filter((e) => e.creator.indexOf(user.email) !== -1));
```
Comment (bonnieli) [inline client/src/pages/Profile/Profile.tsx:73]: maybe make a note of here that filtering won't be the best choice here to do on the frontend. when we have real data, it should be done on the backend!

File: client/src/pages/Profile/useStyles.ts [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+      },
+    },
+    primary: {
+      main: 'black[500]',
+    },
+  },
+  Tab: {
+    backgroundColor: 'transparent',
+  },
+  link: {
```
Comment (bonnieli) [inline client/src/pages/Profile/useStyles.ts:48]: looks like these classnames are a mix of camelCase and PascalCase - i would stay consistent with one convention

Comment (kqpham) [reply client/src/pages/Profile/useStyles.ts:48]: I will fix this right away !

File: client/src/pages/Profile/Profile.tsx (hunk 3 above)
Comment (kqpham) [reply client/src/pages/Profile/Profile.tsx:73]: You are absolutely right I will append that

File: client/src/components/ListView/ListView.tsx (hunk 1 above)
Comment (kqpham) [reply client/src/components/ListView/ListView.tsx:37]: absolutely not I will remove it right away !

File: client/src/pages/Profile/useStyles.ts (hunk 4 above)
Comment (WhoaNellie) [reply client/src/pages/Profile/useStyles.ts:48]: Looks like these haven't been changed yet, but after you do it's good to merge!

**Pull request merged**